tests/integration_tests/end_to_end_test.py

```diff
@@ -23,6 +24,16 @@ def tmp_dir(tmpdir_factory):
     return tmpdir_factory.mktemp('integration_test')
 
 
+mnist_data = None
+
+
+def get_data():
```

Comment:
What's the motivation for this setup?

Reply:
So that each parametrized version of the test doesn't have to reload the data